db/fetcher/encoded_doc.go

```diff
@@ -27,6 +28,10 @@ type encProperty struct {
 	Desc client.FieldDescription
 	Raw  []byte
 
+	// Filter flag to determine if this flag
+	// is needed for eager filter evaluation
+	Filter bool
```

Comment:
question : don't we want to stick to a "normal" convention of naming `bool` vars with prefixes like "Is", "Has"... This one could be names `ShouldFilter`
Current name is surely shorted, but reading `if prop.Filter {` feels a bit ambiguous (at least to my view)
Maybe it's still my C++ vision where you can if basically any type :)